packages/apps/reputation-oracle/server/src/common/utils/hcaptcha.ts

```diff
@@ -1,27 +1,106 @@
 import axios from 'axios';
+import {
+  hCaptchaGetLabeler,
+  hCaptchaRegisterLabeler,
+  hCaptchaVerifyToken,
+} from '../dto/hcaptcha';
 
-export async function verifyToken(
-  url: string,
-  sitekey: string,
-  secret: string,
-  token: string,
-  ip?: string,
-) {
-  const queryParams: any = {
-    secret,
-    sitekey,
-    response: token,
-  };
-
-  if (ip) {
-    queryParams.remoteip = ip;
+/**
+ * Verifies the hCaptcha token.
+ * @param {hCaptchaVerifyToken} data - The data required for token verification.
+ * @returns {Promise<any>} - The verification result.
+ */
+export async function verifyToken(data: hCaptchaVerifyToken): Promise<any> {
+  try {
+    const { url, secret, sitekey, ip, token } = data;
+
+    const queryParams: any = {
+      secret,
+      sitekey,
+      response: token,
+    };
+
+    if (ip) {
+      queryParams.remoteip = ip;
+    }
+
+    const response = await axios.post(
+      `${url}/siteverify`,
+      {},
+      { params: queryParams },
+    );
```

Comment:
To maintain the coherence of the code, why don't you create an `hcaptcha` module and inject the `httpService` instance like we have in `KycModule` and some others in Job Launcher instead of using `axios` here. Same for the others methods

Reply:
It's a good idea to use httpService. Concerning about creating the hCaptcha module, we only use this utility to send requests to a third party service (hCaptcha), it does not have any dependency on Reputation Oracle services. I see some benefits from the integration: we can use authConfig directly instead of passing this data as parameters.

@portuu3 what do you think, integrate the hcaptcha as a module or just integrate httpService to the hcaptcha util?

Reply:
I think it can be implemented as a module to `third-party` folder as this module does not affect to other services in Reputation Oracle.

Reply:
For me it is better to use a module rather than the util. We could do as in human app and have the integrations folder for third party integrations. 
![image](https://github.com/humanprotocol/human-protocol/assets/61605646/27861418-ed74-401b-92f5-d75f9eb81a25)

Reply:
Okay, sounds good for me, I will implement

Reply:
Implemented